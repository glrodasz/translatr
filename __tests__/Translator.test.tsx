import { expect, it, describe } from "vitest";
import { render } from "@testing-library/react";
import Translator from "../pages/index";

describe("[ pages / Translator (index) ]", () => {
  describe("when Translator is rendered", () => {
    it("should render", () => {
      const { asFragment } = render(<Translator />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
