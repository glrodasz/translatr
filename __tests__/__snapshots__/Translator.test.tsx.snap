// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[ pages / Translator (index) ] > when Translator is rendered > should render 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen bg-gray-100 dark:bg-gray-900 text-gray-900 dark:text-gray-100"
  >
    <div
      class="container mx-auto p-4 min-h-screen flex items-center justify-center"
    >
      <div
        class="grid grid-cols-1 md:grid-cols-2 gap-4"
      >
        <div
          class="bg-white dark:bg-gray-800 shadow-md p-4 rounded md:min-w-[300px]"
        >
          <div
            class="flex items-center space-x-4"
          >
            <button
              class="text-sm font-semibold text-gray-500 dark:text-gray-300 py-2 px-4 border-b-2 border-blue-500"
            >
              Detect Language
            </button>
            <button
              class="text-sm font-semibold text-gray-500 dark:text-gray-300 py-2 px-4 "
            >
              Spanish
            </button>
            <button
              class="text-sm font-semibold text-gray-500 dark:text-gray-300 py-2 px-4 "
            >
              English
            </button>
            <button
              class="text-sm font-semibold text-gray-500 dark:text-gray-300 py-2 px-4 "
            >
              Swedish
            </button>
            <span
              class="ml-auto"
            >
              <i
                class="fas fa-exchange-alt text-gray-500 dark:text-gray-300"
              />
            </span>
          </div>
          <textarea
            class="w-full mt-4 p-4 bg-gray-100 dark:bg-gray-700 text-gray-900 dark:text-gray-100 rounded resize-none"
            placeholder="Enter your text here..."
            rows="10"
          />
          <div
            class="h-6 text-right mt-2 text-sm text-gray-500 dark:text-gray-300"
          >
            0 / 100
          </div>
        </div>
        <div
          class="bg-white dark:bg-gray-800 shadow-md p-4 rounded md:min-w-[300px]"
        >
          <div
            class="flex items-center space-x-4"
          >
            <button
              class="text-sm font-semibold text-gray-500 dark:text-gray-300 py-2 px-4 border-b-2 border-blue-500"
            >
              Spanish
            </button>
            <button
              class="text-sm font-semibold text-gray-500 dark:text-gray-300 py-2 px-4 "
            >
              English
            </button>
            <button
              class="text-sm font-semibold text-gray-500 dark:text-gray-300 py-2 px-4 "
            >
              Swedish
            </button>
          </div>
          <textarea
            class="w-full mt-4 p-4 bg-gray-100 dark:bg-gray-700 text-gray-900 dark:text-gray-100 rounded resize-none"
            placeholder="Translation will appear here..."
            readonly=""
            rows="10"
          />
          <div
            class="h-6"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
